samples/manageServices.py

```diff
@@ -0,0 +1,262 @@
+    if actionData['TYPE'] == 'COMMANDS':
+        for i in range(len(actionData['COMMANDS'])):
+            for j in range(len(actionData['COMMANDS'][i])):
+                while 'VM_USERNAME' in actionData['COMMANDS'][i][j]:
+                    actionData['COMMANDS'][i][j] = actionData['COMMANDS'][i][j].replace("VM_USERNAME", username)
+                    
+def replacePassword(actionData, password):
+    if actionData['TYPE'] == 'COMMANDS':
+        for i in range(len(actionData['COMMANDS'])):
+            for j in range(len(actionData['COMMANDS'][i])):
+                while 'VM_PASSWORD' in actionData['COMMANDS'][i][j]:
+                    actionData['COMMANDS'][i][j] = actionData['COMMANDS'][i][j].replace("VM_PASSWORD", password)
+
+def runCommands(vmObject, actionData):
+    for command in actionData['COMMANDS']:
+        for i in range(5):
+            retVal = True
+            try:
+                if vmObject.runCmdOnGuest(command) == False:
+                    retVal = False
+            except vim.fault.InvalidState:
+                retVal = False
+                continue
+            break
+    return retVal
+
+def runScript(vmObject, actionData):
+    localScriptName = actionData['FILENAME']
+    interpreter = actionData['INTERPRETER']
+    remoteScriptName = actionData['UPLOAD_DIR'] + "\\" + localScriptName.split('/')[-1]
+    retVal = False
+    try:
+        retVal = vmObject.uploadAndRun(localScriptName, remoteScriptName, interpreter, True)
+    except Exception as e:
+        print("CAUGHT EXCEPTION: " + str(e))
+        retVal = False
+    return retVal
+
+def checkSuccess(vmObject, actionData):
+    if 'SUCCESS_TYPE' in actionData and 'SUCCESS_METRIC' in actionData:
+        if actionData['SUCCESS_TYPE'] == 'PROCESS':
+            retVal = sampleLib.checkForProcess(vmObject, actionData['SUCCESS_METRIC'])
+    else:
+        print("NO SUCCESS_TYPE OR SUCCESS METRIC IN THE DICTIONARY")
+        retVal = False
+    return retVal
+
+def executeAction(vmObject, actionData):
+    if 'WAIT_SECONDS' in actionData:
+        scheduleDelay = actionData['WAIT_SECONDS']
+    else:
+        scheduleDelay = 30
+    vmObject.powerOn()
+    vmReady = False
+    retVal = True
+    while vmReady is False:
+        vmReady = vmObject.waitForVmToBoot()
+    time.sleep(10)
+    if actionData['TYPE'] == "COMMANDS":
+        retVal = runCommands(vmObject, actionData)
+    if actionData['TYPE'] == "SCRIPT":
+        retVal = runScript(vmObject, actionData)
+    time.sleep(scheduleDelay)
+    if 'SUCCESS_TYPE' in actionData and 'SUCCESS_METRIC' in actionData:
+        retVal = checkSuccess(vmObject, actionData)
+    vmObject.vmObject.ShutdownGuest()
+    time.sleep(10)
+    vmObject.powerOff()
+    return retVal
+    
+def parallelRun(serverConfig, vmName, username, password, actionData, snapshotName):
+    """
+    CREATE SERVER
+    """
+    logFile = './logs/service_management_' + vmName + '.log'
+    try:
+        os.remove(logFile)
```

Comment:
This approach to logging does not sit well for me.   Removing log data at the start of a task to me is a poor experience.

The current log file is adding timestamps for each log entry so I don't think the file should be deleted.  It would be reasonable to just add a log line as a marker for a new parallelRun task that is already below.

Reply:
Not thrilled about it, but I can see your point.  That's probably a less bad way to go.